### ISSUE TITLE
Failed sending data to the peer with latest nightly

### ISSUE THREAD

**Issue by fourbytes** [rust-lang/cargo#9697]
**Problem**

Builds inside docker containers (CI) seem to fail on nightly versions > 2021-07-05.
```
error: failed to compile `sccache v0.2.15`, intermediate artifacts can be found at `/tmp/cargo-installYaFAmQ`

Caused by:
failed to download from `https://crates.io/api/v1/crates/async-io/1.6.0/download`

Caused by:
[55] Failed sending data to the peer (Connection died, tried 5 times before giving up)
```

**Steps**

1. Build a project inside a docker container.

**Possible Solution(s)**

Using nightly 2021-07-05 with `rustup default nightly-2021-07-05` appears to fix the issue.

**Notes**
May be related to #6513.
https://github.community/t/failed-sending-data-to-the-peer-connection-died-tried-5-times-before-giving-up/189130

Output of `cargo version`:
`cargo 1.55.0-nightly (3ebb5f15a 2021-07-02)` (oddly this is the same version that seems to fail in the latest nightly builds).

**Comment by hi-rustin** [comment]
Is it a duplicate of https://github.com/rust-lang/crates.io/issues/3782?

**Comment by fourbytes** [comment]
> Is it a duplicate of [rust-lang/crates.io#3782](https://github.com/rust-lang/crates.io/issues/3782)?

Yep, apologies, searched for the error but didn't see any results.

**Comment by ehuss** [comment]
The next nightly (in about 12 hours from the time of this post) should have a temporary workaround in #9695.

**Comment by weihanglo** [comment]
Looks like this has been fixed in curl <https://github.com/curl/curl/issues/7400>. Going to close as resolved.
